packages/core/src/mpi/inbound-patient-mpi-metriport-api.ts

```diff
@@ -22,10 +22,13 @@ export class InboundMpiMetriportApi implements MPI {
         genderAtBirth: normalizedPatientDemo.genderAtBirth,
       },
     });
+
+    const patientsThatAreOptedIn = foundPatients.filter(patient => !patient.hieOptOut);
+
     const matchingPatients = matchPatients(
       epicMatchingAlgorithm,
       [],
-      foundPatients.map(patientToPatientMPI),
+      patientsThatAreOptedIn.map(patientToPatientMPI),
```

Comment:
So i considered [updating PatientMPI](https://github.com/metriport/metriport/blob/develop/packages/core/src/mpi/shared.ts) to include hieOptOut so that in the function [processInboundXcpd](https://github.com/metriport/metriport/blob/develop/packages/core/src/external/carequality/pd/process-inbound-pd.ts) I could check if the matchingPatient is opted out or not. 

Ran issue where Pick<Patient, "id" | "hieOptOut"> was preserving the optional nature of hieOptOut from PatientCreate, but TypeScript's exactOptionalPropertyTypes flag was making the type checking more strict. So instead of doing 

Pick<Patient, "id"> & {
  hieOptOut?: boolean;
} & PatientData;

I opted for the logic above which is probably cleaner and touches less code.